### PR TITLE
SNOW-594137: Bumped up PythonConnector PATCH version from 2.7.7 to 2.…

### PR DESCRIPTION
**Pull request by sfc-gh-kdama**
…7.8 (#1151)

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]
@sfc-gh-mheavin to review what I wrote

**Review by sfc-gh-mkeller — APPROVED** [review]

File: DESCRIPTION.md [hunk 1]
```diff
@@ -12,18 +12,14 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
-   - Fix the error message when pandas is not installed 
-   -  Implement callproc 
-   -  Ignore content-encoding when GET
-   - handle s3 accelerate configuration correctly 
-   -  update cla_bot repository name
-   - add kwargs to executemany
-   -  Jira creation adjustments
-   -  Set application name based on streamlit being in sys.modules
-   - Update pyopenssl requirement from <22.0.0,>=16.2.0 to >=16.2.0,<23.0.0
+   - Update PyPi documentation link to python specific main page
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:15]: Please be consistent with tense (Update, Fixed)

File: DESCRIPTION.md [hunk 2]
```diff
@@ -12,18 +12,14 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
-   -  Jira creation adjustments
-   -  Set application name based on streamlit being in sys.modules
-   - Update pyopenssl requirement from <22.0.0,>=16.2.0 to >=16.2.0,<23.0.0
+   - Update PyPi documentation link to python specific main page 
+   - Fixed an error message that appears when pandas optional dependency group is required but is not installed
+   - Implemented callproc method from DB API 2
+   - Fixed a bug where decryption took place before decompression when downloading files from stages
+   - Fixed a bug where we handled s3 accelerate configuration incorrectly 
+   - Executemany now passes extra named-arguments to execute
+   - Application name now gets set to streamlit if streamlit is imported and application name was not explicitly set
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:21]: Automatically sets the application name to streamlit when ...

File: DESCRIPTION.md [hunk 3]
```diff
@@ -12,18 +12,14 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
+   - Bumped pyopenssl dependency version
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:22]: to what version?

File: DESCRIPTION.md [hunk 4]
```diff
@@ -12,18 +12,14 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
-   -  update cla_bot repository name
-   - add kwargs to executemany
-   -  Jira creation adjustments
-   -  Set application name based on streamlit being in sys.modules
-   - Update pyopenssl requirement from <22.0.0,>=16.2.0 to >=16.2.0,<23.0.0
+   - Update PyPi documentation link to python specific main page 
+   - Fixed an error message that appears when pandas optional dependency group is required but is not installed
+   - Implemented callproc method from DB API 2
+   - Fixed a bug where decryption took place before decompression when downloading files from stages
+   - Fixed a bug where we handled s3 accelerate configuration incorrectly
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:19]: Avoid "we"

File: DESCRIPTION.md [hunk 5]
```diff
@@ -12,18 +12,14 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
-   - add kwargs to executemany
-   -  Jira creation adjustments
-   -  Set application name based on streamlit being in sys.modules
-   - Update pyopenssl requirement from <22.0.0,>=16.2.0 to >=16.2.0,<23.0.0
+   - Update PyPi documentation link to python specific main page 
+   - Fixed an error message that appears when pandas optional dependency group is required but is not installed
+   - Implemented callproc method from DB API 2
+   - Fixed a bug where decryption took place before decompression when downloading files from stages
+   - Fixed a bug where we handled s3 accelerate configuration incorrectly 
+   - Executemany now passes extra named-arguments to execute
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:20]: Is Executemany really supposed to be one word? Why is named-arguments hyphenated?

File: DESCRIPTION.md [hunk 6]
```diff
@@ -12,18 +12,14 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
-   -  Ignore content-encoding when GET
-   - handle s3 accelerate configuration correctly 
-   -  update cla_bot repository name
-   - add kwargs to executemany
-   -  Jira creation adjustments
-   -  Set application name based on streamlit being in sys.modules
-   - Update pyopenssl requirement from <22.0.0,>=16.2.0 to >=16.2.0,<23.0.0
+   - Update PyPi documentation link to python specific main page 
+   - Fixed an error message that appears when pandas optional dependency group is required but is not installed
+   - Implemented callproc method from DB API 2
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:17]: Implemented the DB API 2 callproc method [should callproc by callproc() to reflect it is a method?]

File: DESCRIPTION.md (hunk 3 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:22]: `>=16.2.0,<23.0.0`

File: DESCRIPTION.md (hunk 1 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:15]:
```suggestion
   - Updated PyPi documentation link to python specific main page 
```

File: DESCRIPTION.md (hunk 6 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:17]:
```suggestion
   - Implemented the DB API 2 callproc() method
```

File: DESCRIPTION.md (hunk 4 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:19]:
```suggestion
   - Fixed a bug where s3 accelerate configuration was handled incorrectly 
```

File: DESCRIPTION.md (hunk 5 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:20]:
```suggestion
   - Extra named arguments given executemany() are now forwarded to execute()
```

File: DESCRIPTION.md (hunk 2 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:21]:
```suggestion
   - Automatically sets the application name to streamlit when streamlit is imported and application name was not explicitly set
```

File: DESCRIPTION.md (hunk 3 above)
Comment (sfc-gh-mkeller) [reply DESCRIPTION.md:22]:
```suggestion
   - Bumped pyopenssl dependency version to >=16.2.0,<23.0.0
```

**Review by sfc-gh-mkeller — COMMENTED** [review]
@sfc-gh-mheavin do these work?

**Review by sfc-gh-mheavin — APPROVED** [review]

**Pull request merged**